### ISSUE TITLE
what is the best way to tune up active connections

### ISSUE THREAD

**Issue by jjhesk** [gobwas/ws#68]
what is the max count for active connection given a specific linux config.
i used this https://docs.oracle.com/cd/E19623-01/820-6168/file-descriptor-requirements.html
1. set 65535 as limit

**Comment by blackbone** [comment]
>
>
>     1. set 65535 as limit

65535 is a maximum port count in a net stack. keep in mind that ports 0-1023 are reserved in system also as part of ports can be used by other applications. So, in my experience 50k active connections limit per host is more than you exacally need (you need a very powerful server to handle it). If you got that limit - it's time to think about multi host architecture, load balancing, etc.

**Comment by gobwas** [comment]
@blackbone no, that is incorrect. 65535 (2^16 - 1) is the maximum number of connections between **two** hosts. That is, single tcp socket identified by the so called four-tuple – source host, source port, destination host and destination port. So server host and port are the same, client host too; thus – single client can establish up to 65K connections to single host:port. 

What is really @jjhesk is asking about is about **file descriptors** needed for each socket. This could be configured somehow for any particular OS.

TL;DR about that article: seems like it changing fd limit from 1024 to 65K.
